feat(sbb-badge): provide badge styling via attribute (#2639)
sbb-design-systems/lyne-components: src/storybook/styles/badge/readme.md

## Badge

The `sbb-badge` attribute can be applied to every element, but is mainly designed to be placed on an `sbb-icon`.

### API

| CSS attribute name   | Type              | Description                                                                  |
| -------------------- | ----------------- | ---------------------------------------------------------------------------- |
| `sbb-badge`          | `string`          | Content displayed in badge.                                                  |
| `sbb-badge-position` | `before \| after` | Positioning of the badge relative to the element where the badge is applied. |

Moreover, we provide the Sass mixin `badge` which contains the badge styling for general usage.

### Usage

> #### standard / after

Reviewer: ```suggestion
#### Position default/after
```

```html
<sbb-icon icon-name="controls-small" sbb-badge="2"></sbb-icon>
<sbb-screen-reader-only>Currently 2 settings changed.</sbb-screen-reader-only>
```

> #### before

Reviewer: ```suggestion
#### Position before
```

```html
<sbb-icon icon-name="controls-small" sbb-badge="2" sbb-badge-position="before"></sbb-icon>
<sbb-screen-reader-only>Currently 2 settings changed.</sbb-screen-reader-only>
```

### Accessibility

For screen readers it's important to output the meaning of the badge. This attribute itself doesn't

> output anything to screen readers. Setting an `aria-label`s or the `<sbb-screen-reader-only>` element can be helpful.

Reviewer: ```suggestion
output anything to screen readers. Setting an `aria-label` attribute on the same element or adding
`<sbb-screen-reader-only>` inside the element can be used to provide the relevant information.
```
